src/commands/delete.js

```diff
@@ -5,13 +5,22 @@ const Application = require('../models/application.js');
 const Logger = require('../logger.js');
 
 async function deleteApp (params) {
-  const { alias, yes: skipConfirmation } = params.options;
-  const appDetails = await AppConfig.getAppDetails({ alias });
+  const { alias, app: appIdOrName, yes: skipConfirmation } = params.options;
+  const { ownerId, appId } = await Application.resolveId(appIdOrName, alias);
 
-  await Application.deleteApp(appDetails, skipConfirmation);
-  await Application.unlinkRepo(appDetails.alias);
-
-  Logger.println('The application has been deleted');
+  const app = await Application.get(ownerId, appId);
+  if (app == null) {
+    Logger.println('The application doesn\'t exist');
+  }
+  else {
+    // delete app
+    await Application.deleteApp(app, skipConfirmation);
+    Logger.println('The application has been deleted');
+    // unlink app
+    if (await AppConfig.removeLinkedApplication({ appId, alias })) {
+      Logger.println('The application has been unlinked');
+    }
+  }
 };
```

Comment:
praise: nice improvement!

Reply:
nitpick: can we move the result of the async function outside of the `if` in a variable (maybe `wasUnlinked`)?